src/app.ts

```diff
@@ -72,6 +76,27 @@ export function showSidebar() {
   SpreadsheetApp.getUi().showSidebar(html);
 }
 
+export function fetchContextByItemId(itemId: string) {
```

Comment:
Could you please modify this method to not have any input params? Instead:
INPUT
In the sheet you can provide a comma-separated list of items. Define that input value (row/col) in config.ts and read this input here. Split it and process it as a set (to automatically remove any duplicates erroneously entered by the user)
OUTPUT
a map of input itemId to context JSON string. If the sheet expects the output as a 2D String array then please return it as such. The output should automatically be written to the expected location (Config sheet rows 14 onwards)

We could therefore trigger this method to fire at any point during the loading process (when the user opens the FeedGen sidebar for example). So basically the user starts with the "Configure" sheet, enters all their details, then "Launches FeedGen" which triggers this method to prefill the prompt context. Alternatively, we add a dedicated button to the sidebar called "Configure" which does this. Please discuss with @paranerd

Reply:
as discussed, changing name to FEEDGEN_CREATE_JSON_CONTEXT_FOR_ITEM